src/main/java/fern/analysis/ShortestPath.java

```diff
@@ -202,7 +202,7 @@ public ShortestPathAction(NodeChecker checker) {
      * @return
      */
     public Path getPath(int species) {
-      LinkedList<Integer> path = new LinkedList<Integer>();
```

Comment:
I like the explicit mention type when initializing. Makes things little easy to follow. I'd prefer keeping this the original way.

Reply:
Fair! I personally find Java verbose enough with all the _necessary_ declaration, so I tend to trim the unnecessary ones as much as possible - it seems clear here even in the less explicitly typed version that a `LinkedList` containing `Integer`s is being created. But tastes differ!